### ISSUE TITLE
Add support for shadows

### ISSUE THREAD

**Issue by Iss-in** [dunst-project/dunst#926]
Now most wayland compositors  dont implement shadows themselves, instead allowing the client window to draw them for themselves. Will it be possible to implement this in dunst itself ?

**Comment by wael444** [comment]
yes but it might take plenty of time and code, i suggest you just use a compositor instead of adding support directly into the software

**Comment by fwSmit** [comment]
> yes but it might take plenty of time and code, i suggest you just use a compositor instead of adding support directly into the software

Agreed. But I'd probably accept a PR if it doesn't add too much complexity.

**Comment by Iss-in** [comment]
most wayland compositors, especially sway aren't drawing up border shadows anymore, and are letting it to be part of csd itself. this also includes most gtk apps ( they all draws shadows for themselves ) including firefox on wayland.

implementation is quite simple imo, shadow is just a black border with progressive transparency (alpha gradient) applied to it. should be really easy to do in cairo, sadly I just dont have enough experience with it . hopefully someone will pick it up

**Comment by traeanto** [comment]
please mention me if box shadow has been added

**Comment by apprehensions** [comment]
No

**Comment by bynect** [comment]
> most wayland compositors, especially sway aren't drawing up border shadows anymore, and are letting it to be part of csd itself. this also includes most gtk apps ( they all draws shadows for themselves ) including firefox on wayland.
> 
> implementation is quite simple imo, shadow is just a black border with progressive transparency (alpha gradient) applied to it. should be really easy to do in cairo, sadly I just dont have enough experience with it . hopefully someone will pick it up

do you have any example on hand for implementing shadows? I may work on that in my effort to make dunst more aesthetic, but I am a little short on time right now so any pointer is appreciated
